routes/nft.go

```diff
@@ -694,14 +693,53 @@ func (fes *APIServer) GetNFTShowcase(ww http.ResponseWriter, req *http.Request)
 		postEntryResponse, err := fes._postEntryToResponse(
 			postEntry, false, fes.Params, utxoView, readerPublicKeyBytes, 2)
 		if err != nil {
-			_AddInternalServerError(ww, fmt.Sprint("GetNFTShowcase: Found invalid post entry for NFT hash."))
-			return
+			return nil, fmt.Errorf(
+				"GetNFTShowcase: Found invalid post entry for NFT hash.")
 		}
 
 		nftCollectionResponse := fes._nftEntryToNFTCollectionResponse(nftEntry, postEntry.PosterPublicKey, postEntryResponse, utxoView, verifiedMap, readerPKID)
 		nftCollectionResponses = append(nftCollectionResponses, nftCollectionResponse)
 	}
+	return nftCollectionResponses, nil
+}
 
+func (fes *APIServer) GetCurrentNFTShowcase(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetNFTShowcaseRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetNFTShowcase: Error parsing request body: %v", err))
+		return
+	}
+	fes.GetNFTShowcase(ww, req, -1, requestData.ReaderPublicKeyBase58Check)
+}
+
+// If dropIdx == -1, get current drop
+func (fes *APIServer) GetNFTShowcase(ww http.ResponseWriter, req *http.Request, dropIdx int, readerPublicKeyBase58Check string) {
```

Comment:
do we need `req *http.Request` as a param in this function?  doesn't look like it's used.